dlc-manager/src/manager.rs

```diff
@@ -271,19 +272,35 @@ where
 
     /// Function called to create a new DLC. The offered contract will be stored
     /// and an OfferDlc message returned.
+    ///
+    /// This function will fetch the oracle announcements from the oracle.
     pub fn send_offer(
         &mut self,
         contract_input: &ContractInput,
         counter_party: PublicKey,
     ) -> Result<OfferDlc, Error> {
-        contract_input.validate()?;
-
         let oracle_announcements = contract_input
             .contract_infos
             .iter()
             .map(|x| self.get_oracle_announcements(&x.oracles))
             .collect::<Result<Vec<_>, Error>>()?;
 
+        self.send_offer_with_announcements(contract_input, counter_party, oracle_announcements)
+    }
+
+    /// Function called to create a new DLC. The offered contract will be stored
+    /// and an OfferDlc message returned.
+    ///
+    /// This function allows to pass the oracle announcements directly instead of
+    /// fetching them from the oracle.
+    pub fn send_offer_with_announcements(
+        &mut self,
+        contract_input: &ContractInput,
+        counter_party: PublicKey,
+        oracle_announcements: Vec<Vec<OracleAnnouncement>>,
+    ) -> Result<OfferDlc, Error> {
+        contract_input.validate()?;
```

Comment:
You removed the validation you had before right? Also this is called within the `offer_contract` method below so it's redundant. Maybe the best would be to validate that the announcements are valid with the contract(s) info (it seems it's actually lacking at the moment even before your change if I'm not mistaken). I'm ok with doing it in another PR also.

Reply:
`contract_input.validate()` was called in `send_offer` before, but looks like that was redundant so will remove